src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -763,16 +763,28 @@ private string GetDebuggerDisplay()
                         AliasSymbol alias;
                         var declType = BindVariableTypeWithAnnotations(component.Designation, diagnostics, component.Type.SkipScoped(out _).SkipRef(out _), ref isConst, out isVar, out alias);
                         Debug.Assert(isVar == !declType.HasType);
-                        if (component.Designation.Kind() == SyntaxKind.ParenthesizedVariableDesignation && !isVar)
+                        if (component.Designation.Kind() == SyntaxKind.ParenthesizedVariableDesignation)
                         {
-                            // An explicit type is not allowed with a parenthesized designation
-                            Error(diagnostics, ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, component.Designation);
+                            if (!isVar)
+                            {
+                                // An explicit type is not allowed with a parenthesized designation
+                                Error(diagnostics, ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, component.Designation);
+                            }
+                            else if (node.Parent is not ArgumentSyntax)
+                            {
+                                // check for use of `var (x, y, z)`.  Only need to report this in a non-argument
+                                // position. If it's an argument, then we have `(int x, var (y, z))` and we will have
+                                // already reported the parent tuple, so no need to report on the inner designation.
+                                MessageID.IDS_FeatureTuples.CheckFeatureAvailability(diagnostics, component.Designation);
```

Comment:
note: the above preserves the logic we had in the parser where we'd report on the outer designatino, but not hte inner ones.  We could simplify this code (by getting rid of the `if (node.Parent is not ArgumentSyntax)` check at the cost of more error messages to the user with recursive designations.

Generally speaking, we do just repeat errors with recursive constructs in other cases.  So up to you @jcouv what you'd prefer here?